foreman.te

```diff
@@ -171,190 +188,156 @@ require{
-dev_read_sysfs(passenger_t)
-dev_search_sysfs(passenger_t)
-dev_read_rand(passenger_t)
+# Allow Foreman to do DNS queries via Ruby stdlib net package
+# http://projects.theforeman.org/issues/8030
+corenet_udp_bind_generic_port(foreman_rails_t)
 
-# Passenger 6 is setting socket context explicitly in
-# https://github.com/phusion/passenger/blob/master/src/agent/Core/CoreMain.cpp#L376-L406
-allow passenger_t self:process setsockcreate;
-
-# Passenger 6 reads and writes in /etc/pki/nssdb
-miscfiles_manage_cert_dirs(passenger_t)
-miscfiles_manage_generic_cert_files(passenger_t)
-
-# Passenger 6 compiles and executes during start file named
-# /tmp/passenger-native-support-XXXXXX/passenger_native_support.so
-can_exec(passenger_t, passenger_tmp_t)
-
-# Logrotate in RHEL7 does not allow signals, RHBZ#1527522
-gen_require(`
-    type logrotate_t;
-')
-systemd_config_generic_services(logrotate_t)
-
-optional_policy(`
-    tunable_policy(`passenger_run_foreman', `
-        admin_pattern(httpd_t, foreman_lib_t, foreman_lib_t)
-        admin_pattern(passenger_t, foreman_lib_t, foreman_lib_t)
-        admin_pattern(passenger_t, foreman_var_run_t, foreman_var_run_t)
-
-        # Allow Foreman to connect to hosts and guests
-        corenet_tcp_connect_virt_port(passenger_t)
-        corenet_tcp_connect_ssh_port(passenger_t)
-
-        # Allow Foreman to write to the SQlite databases
-        read_files_pattern(passenger_t, foreman_db_t, foreman_db_t)
-        write_files_pattern(passenger_t, foreman_db_t, foreman_db_t)
+# rubygem FFI is used in foreman-tasks (BZ#1670109 / RM#26951)
+allow foreman_rails_t self:process execmem;
 
-        # Allow Foreman to do DNS queries via Ruby stdlib net package
-        # http://projects.theforeman.org/issues/8030
-        corenet_udp_bind_generic_port(passenger_t)
+# Connecting to Foreman Proxy on port 9090 (Katello)
+# or to remote Cockpit instance (via Remote Execution)
+allow foreman_rails_t websm_port_t:tcp_socket name_connect;
 
-        admin_pattern(passenger_t, passenger_tmp_t, passenger_tmp_t)
-    ')
-')
+# Connecting to Foreman Proxy on a defined port
+allow foreman_rails_t foreman_proxy_port_t:tcp_socket name_connect;
 
+# Connecting to PostgreSQL
+corenet_tcp_connect_postgresql_port(foreman_rails_t)
 optional_policy(`
-    tunable_policy(`passenger_run_foreman', `
-        allow passenger_t self:process getsession;
-        fs_rw_anon_inodefs_files(passenger_t)
-        allow passenger_t httpd_t:unix_stream_socket { read write getattr };
-        fs_getattr_xattr_fs(passenger_t)
-        allow passenger_t self:capability2 block_suspend;
-    ')
+    postgresql_stream_connect(foreman_rails_t)
 ')
 
-tunable_policy(`httpd_run_foreman', `
-    admin_pattern(httpd_t, passenger_tmp_t, passenger_tmp_t)
-    admin_pattern(passenger_t, httpd_tmp_t, httpd_tmp_t)
-')
+# Connecting to Redis
+corenet_tcp_connect_redis_port(foreman_rails_t)
 
+# Connecting to puppet server
```

Comment:
Foreman never connects to the puppetmaster. Only foreman-proxy does

Reply:
That was probably puppet master: https://projects.theforeman.org/issues/21887

It was passenger's infra preloading the app. Removed.